homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -179,6 +181,57 @@ def get(self, request, username):
         return self.json(json_response)
 
 
+class HueFullStateView(HomeAssistantView):
+    """Return full state view of emulated hue."""
+
+    url = "/api/{username}"
+    name = "emulated_hue:username:lights:state"
+    requires_auth = False
+
+    def __init__(self, config):
+        """Initialize the instance of the view."""
+        self.config = config
+
+    @core.callback
+    def get(self, request, username):
+        """Process a request to get the list of available lights."""
+        if not is_local(request[KEY_REAL_IP]):
+            return self.json_message("only local IPs allowed", HTTP_BAD_REQUEST)
+        if username not in HUE_API_USERNAME:
+            json_response = [
+                {
+                    "error": {
+                        "address": "/",
+                        "description": "unauthorized user",
+                        "type": "1",
+                    }
+                }
+            ]
+        else:
+            hass = request.app["hass"]
+            json_response = {}
+
+            for entity in hass.states.async_all():
```

Comment:
We should make reusable logic for the configuration so that the sub-views use the same logic as the fetch-all view to fill in the different parts.